### PR TITLE
Add UserWardrobeManager

### PR DESCRIPTION
**Pull request by chrismpettyjohn**
**Story**
As a Developer, I want a reusable service for adding and removing looks from the user wardrobe.

**Changes**
1. Created  UserWardrobeManager
2. Updated SaveWardrobeOutfitEvent to use the new UserWardrobeManager

**Issues**
Related to #4 

**Testing**
I validated locally that users could create outfits and save them in various slots, or overwrite existing slots.

**Video**
Nitro

### CODE REVIEW

File: Communication/Packets/Incoming/Avatar/SaveWardrobeOutfitEvent.cs [hunk 1]
```diff
@@ -25,22 +25,8 @@ public Task Parse(GameClient session, ClientPacket packet)
         var gender = packet.PopString();
         look = _figureDataManager.ProcessFigure(look, gender, session.GetHabbo().GetClothing().GetClothingParts, true);
 
-        using (var connection = _database.Connection())
-        {
-            int rows = connection.Execute("SELECT null FROM `user_wardrobe` WHERE `user_id` = @id AND `slot_id` = @slot",
-                new { id = session.GetHabbo().Id, slot = slotId });
+        _userWardrobeManager.SaveOutfit(session.GetHabbo().Id, slotId, gender, look);
```
Comment (80O) [inline Communication/Packets/Incoming/Avatar/SaveWardrobeOutfitEvent.cs:28]: Task needs to be awaited. Add await in front and make method async.

File: Plus Emulator.csproj [hunk 2]
```diff
@@ -29,6 +29,9 @@
 	  </Reference>
 	</ItemGroup>
 
+	<ItemGroup>
```
Comment (80O) [inline Plus Emulator.csproj:32]: Itemgroup is not needed.

Comment (chrismpettyjohn) [reply Plus Emulator.csproj:32]: My IDE automatically did this, will remove

File: Communication/Packets/Incoming/Avatar/SaveWardrobeOutfitEvent.cs (hunk 1 above)
Comment (chrismpettyjohn) [reply Communication/Packets/Incoming/Avatar/SaveWardrobeOutfitEvent.cs]:
```
public interface IPacketEvent
{
    Task Parse(GameClient session, ClientPacket packet);
}
```

That would require updating the underlying interface for packet events.   I made an alternative change, however that may do the trick.

```
        Task.Run(async () => await _userWardrobeManager.SaveOutfit(session.GetHabbo().Id, slotId, gender, look));

```

Comment (chrismpettyjohn) [reply Communication/Packets/Incoming/Avatar/SaveWardrobeOutfitEvent.cs]: Updated after a misunderstanding 👍

File: Plus Emulator.csproj (hunk 2 above)
Comment (chrismpettyjohn) [reply Plus Emulator.csproj]: Pushed

**Comment by chrismpettyjohn** [comment]
Upon testing, it looks like there's a few issues with the DB queries to work out.  Will update sometime tomorrow

**Comment by chrismpettyjohn** [comment]
I resolved the test issues and have verified users can successfully interact with their wardrobe.  Apart of these changes, are a new database migration that prevents multiple rows being added for the same user, slot and gender combination

Nitro

**Comment by 80O** [comment]
I think unique constraint should be over user_id and slot_id. Cant insert multiple looks for different genders in the same slot I think?

**Comment by chrismpettyjohn** [comment]
> I think unique constraint should be over user_id and slot_id. Cant insert multiple looks for different genders in the same slot I think?

Changes pushed

File: HabboHotel/Users/Wardrobe/UserWardrobeManager.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    }
+            );
+
+        }
+
+		public async Task RemoveOutfit(int userID, int slotID, string gender)
+        {
+            using var connection = _database.Connection();
+            await connection.ExecuteAsync(
+               "DELETE FROM `user_wardrobe` WHERE `user_id` = @userID AND `slot_id` = @slotID AND `gender` = @gender",
```
Comment (80O) [inline HabboHotel/Users/Wardrobe/UserWardrobeManager.cs:36]: I dont think Gender is needed to delete. Just slotId

**Comment by chrismpettyjohn** [comment]
Closing as resolved with more fn by above PR

**Pull request closed without merging**